HHS/Head-Start-TTADP
# [TTAHUB-942] Fixes for Objectives

## Description of change

This PR fixes the below bugs:

- Save Goal then use context menu to edit shows both read only and edit screen.
- Fix deleting Objective would cause FK constraint error for 'ActivityReportObjective'.
- If using a existing Objective we correctly remove it from the Objective drop (also when editing via context).


## How to test

Create a goal on a AR that uses existing goals. When an objective is applied it should be removed from the drop down list. Saving the goal and returning to edit should also reflect the used objectives and no longer display both the edit and readonly views.


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-942


## Checklists

### Every PR


~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy


### After merge/deploy
